src/monocle_apptrace/exporters/aws/s3_exporter.py

```diff
@@ -110,7 +110,16 @@ def __serialize_spans(self, spans: Sequence[ReadableSpan]) -> str:
         try:
             # Serialize spans to JSON or any other format you prefer
             span_data_list = [span.to_json() for span in spans]
-            return "[" + ", ".join(span_data_list) + "]"
+            valid_json_list = []
```

Comment:
The below should be enough. Pls check
```
def __serialize_spans(self, spans: Sequence[ReadableSpan]) -> str:
    valid_json_list = []
    for span in spans:
        try:
            valid_json_list.append(json.dumps(span))
        except json.JSONDecodeError as e:
            logger.warn(f"Invalid JSON format in span: {span.id}. Warn: {e}")
            continue
        # Join valid JSON objects into NDJSON format
    ndjson_data = "\n".join(valid_json_list) + "\n"
    return ndjson_data
```

Reply:
not working ,  as span is coming ReadableSpan object , to  serialized this object we are using to_json() method of ReadableSpan class .
here is link - 
https://opentelemetry-python.readthedocs.io/en/latest/_modules/opentelemetry/sdk/trace.html#ReadableSpan.to_json